plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
On Tue, May 17, 2016 at 04:05:46PM -0700, Michael Grosser wrote:
> >      def deployment_labels
> > -      # Deployment labels should not include the ids of the release, role or deploy groups
> > -      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
> > +      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
> 
> we decided on stopping to mess with labels altogether and only add annotations (which will avoid any kind of matching issues), sounds good ?

I think labels are more easy to obtain and query with kubectl (and maybe show in
a UI too).

I think this is becoming messy because there are several labels in the pod and
we are not treating them differently. One thing is the selector (spec.selector),
that uses labels to match which pods should be replaced, and one other is the
labels we use in a spec.template.metadata.labels that the labels there should be
the labels that a pod has when it's been created with this deployment.

We can have project, role and maybe deploy_group in the selector, and add the
revision, tag, etc. in spec.template.metadata.labels and, therefore, it should
be in the pods running as labels.

I think it will be better to have the selector and the labels used for the pods
separated (maybe one is the others plus some, so the selector will for sure
match, and we add stuff about this particular release we want to have as
labels).

What do you think of something like this?

I think it can be pretty clean and have all the advantages of using labels